### PR TITLE
Initial composition list

### PR DESCRIPTION
**Pull request by gassmoeller**
This is based on #1038, and implements the same thing for initial composition plugins (i.e. selecting multiple initial composition plugins, which are added). 
A feature I would like to add later is to select field-specific plugins (i.e. `set List of model names = all: function, lithosphere field: weak seed`), but this is something for another PR.

### CODE REVIEW

File: benchmarks/shear_bands/shear_bands.cc [hunk 1]
```diff
@@ -748,27 +748,18 @@ namespace aspect
...
-          amplitude           = initial_conditions->get_wave_amplitude();
-          initial_band_angle  = initial_conditions->get_initial_band_angle();
-        }
-      else
-        {
-          AssertThrow(false,
-                      ExcMessage("Postprocessor shear bands growth rate only works with the initial conditions model "
-                                 "plane wave melt bands."));
-        }
+      amplitude           = initial_conditions->get_wave_amplitude();
```
Comment (bangerth) [inline benchmarks/shear_bands/shear_bands.cc]: By getting rid of the `if(... == NULL)`, you will get a seg fault here if `initial_conditions` is NULL.

File: benchmarks/solitary_wave/solitary_wave.cc [hunk 2]
```diff
@@ -662,21 +662,13 @@ namespace aspect
...
-          amplitude           = initial_conditions->get_amplitude();
-          background_porosity = initial_conditions->get_background_porosity();
-          offset              = initial_conditions->get_offset();
-        }
-      else
-        {
-          AssertThrow(false,
-                      ExcMessage("Postprocessor Solitary Wave only works with the initial conditions model Solitary wave."));
-        }
+      amplitude           = initial_conditions->get_amplitude();
```
Comment (bangerth) [inline benchmarks/solitary_wave/solitary_wave.cc]: same here

**Comment by gassmoeller** [comment]
I merged #1038, and rebased this one. @bangerth: Ready for review.

File: tests/solitary_wave.cc [hunk 3]
```diff
@@ -1,1031 +1 @@
-#include <aspect/material_model/interface.h>
```
Comment (bangerth) [inline tests/solitary_wave.cc]: nice job undoing the code duplication!

**Comment by bangerth** [comment]
Looks good -- please merge once the tester is happy!

**Pull request merged**